ocpp/v201/datatypes.py

```diff
@@ -0,0 +1,729 @@
+    """
+
+    vendor_name: str
+    model: str
+    modem: Optional[ModemType] = None
+    serial_number: Optional[str] = None
+    firmware_version: Optional[str] = None
+
+
+@dataclass
+class OCSPRequestDataType:
+    """OCSPRequestDataType is used by: AuthorizeRequest, GetCertificateStatusRequest"""
+
+    hash_algorithm: enums.HashAlgorithmType
+    issuer_name_hash: str
+    issuer_key_hash: str
+    serial_number: str
+    responder_url: str
+
+
+@dataclass
+class VariableAttributeType:
+    """
+    Attribute data of a variable.
+    VariableAttributeType is used by: NotifyReportRequest.ReportDataType
+    """
+
+    type: Optional[enums.AttributeType] = None
+    value: Optional[str] = None
+    mutability: Optional[enums.MutabilityType] = None
+    persistent: Optional[bool] = None
+    constant: Optional[bool] = None
+
+
+@dataclass
+class VariableCharacteristicsType:
+    """
+    Fixed read-only parameters of a variable.
+    VariableCharacteristicsType is used by: NotifyReportRequest.ReportDataType
+    """
+
+    data_type: enums.DataType
+    supports_monitoring: bool
+    unit: Optional[str] = None
+    min_limit: Optional[float] = None
+    max_limit: Optional[float] = None
+    values_list: Optional[str] = None
+
+
+@dataclass
+class ReportDataType:
+    """Class to report components, variables and variable attributes and characteristics"""
+
+    component: ComponentType
+    variable: VariableType
+    variable_attribute: VariableAttributeType
+    variable_characteristics: Optional[VariableCharacteristicsType] = None
+
+
+@dataclass
+class SalesTariffType:
+    """
+    This dataType is based on dataTypes from ISO 15118-2.
+    SalesTariffType is used by: ChargingScheduleType
+    """
+
+    id: int
+    sales_tariff_entry: SalesTariffEntryType
+    sales_tariff_description: Optional[str] = None
+    num_e_price_levels: Optional[int] = None
+
+
+@dataclass
+class SetMonitoringDataType:
+    """
+    Class to hold parameters of SetVariableMonitoring request.
+    SetMonitoringDataType is used by: SetVariableMonitoringRequest
+    """
+
+    value: float
+    type: enums.MonitorType
+    severity: int
+    component: ComponentType
+    variable: VariableType
+    id: Optional[int] = None
+    transaction: Optional[bool] = None
+
+
+@dataclass
+class SetMonitoringResultType:
+    """
+    Class to hold result of SetVariableMonitoring request.
+    SetMonitoringResultType is used by: SetVariableMonitoringResponse
+    """
+
+    status: enums.SetMonitoringStatusType
+    type: enums.MonitorType
+    severity: int
+    component: ComponentType
+    variable: VariableType
+    id: Optional[int] = None
+    status_info: Optional[StatusInfoType] = None
+
+
+@dataclass
+class SetVariableDataType:
+    """SetVariableDataType is used by: SetVariablesRequest"""
+
+    attribute_value: str
+    component: ComponentType
+    variable: VariableType
+    attribute_type: Optional[enums.AttributeType] = None
```

Comment:
Even if Auke was not sure on adding the custom data field, it is by default None, so it wont impact nothing if we add it and it will make it in sync with the schemas